src/main/java/nom/tam/util/RandomChannelFileIO.java

```diff
@@ -0,0 +1,175 @@
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ * #L%
+ */
+
+
+import java.io.FileDescriptor;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Concrete implementation of RandomAccessFileIO for FileChannels.  This is a generic implementation using the Java
+ * FileChannel API.
+ * An example of an implementation of the underlying FileChannel is the NIO SPI file system provider to Amazon S3.
+ *
+ * @see <a href="https://github.com/awslabs/aws-java-nio-spi-for-s3">AWS NIO SPI library</a>
+ */
+public class RandomChannelFileIO implements RandomAccessFileIO {
+    private static final Logger LOGGER = LoggerHelper.getLogger(RandomChannelFileIO.class);
+
+    private final long fileSize;
+    private final FileChannel fileChannel;
+
+    public RandomChannelFileIO(final Path filePath) throws IOException {
+        this(filePath, true);
+    }
+
+    public RandomChannelFileIO(final Path filePath, final boolean readOnly) throws IOException {
+        this(Files.size(filePath), FileChannel.open(filePath, readOnly ? new OpenOption[]{StandardOpenOption.READ} :
+                new OpenOption[]{StandardOpenOption.READ, StandardOpenOption.WRITE}));
+    }
+
+    /**
+     * Used for testing purposes.
+     *
+     * @param fileSize    The size (length) of the file in bytes.
+     * @param fileChannel The open channel to the file.
+     */
+    RandomChannelFileIO(long fileSize, FileChannel fileChannel) {
+        this.fileSize = fileSize;
+        this.fileChannel = fileChannel;
+    }
+
+
+    // Read UTF string (mock implementation)
+    @Override
+    public String readUTF() {
+        throw new UnsupportedOperationException("readUTF is not implemented for FileChannels.");
+    }
+
+    @Override
+    public FileChannel getChannel() {
+        return this.fileChannel;
+    }
+
+    @Override
+    public FileDescriptor getFD() {
+        throw new UnsupportedOperationException("FileDescriptor is not implemented for FileChannels.");
+    }
+
+    @Override
+    public void setLength(long l) {
+        throw new UnsupportedOperationException("Setting file length is not implemented for FileChannels.");
+    }
+
+    @Override
+    public void writeUTF(String s) {
+        throw new UnsupportedOperationException("writeUTF operation is not implemented for FileChannels.");
+    }
+
+    @Override
+    public void close() throws IOException {
+        // Just in case (i.e. this is a unit test running), check for null.
+        if (fileChannel != null) {
```

Comment:
I think by the public constructor `fileChannel` cannot be `null`, so no need to check here. Of course, one could in principle concoct a way to access the package-level constructor also, and pass `null` there to initialize `fileChannel`. Maybe the testing constructor could throw an `IllegalArgumentException` if the `fileChannel` argument is `null`.